### PR TITLE
GROUNDWORK-1667 Define status calcs in transit package

### PR DESCRIPTION
**Pull request by ymkins**

### CODE REVIEW

**Review by VladislavSenkevich — APPROVED** [review]

File: transit/transit.go [hunk 1]
```diff
@@ -1017,3 +1017,147 @@ type AgentIdentity struct {
...
+			return ServiceOk
+		} else {
+			if (warning != nil && critical != nil) && float64(value.IntegerValue) >= criticalValue {
+				return ServiceUnscheduledCritical
+			}
+			if (warning != nil && critical != nil) && float64(value.IntegerValue) >= warningValue {
+				return ServiceWarning
+			}
+			return ServiceOk
+		}
```
Comment (Hipska) [inline transit/transit.go:1116]: Lot's of code that has been duplicated. Why can't the value also be converted to float first and then do the checking only once.

File: transit/transit.go [hunk 2]
```diff
@@ -1017,3 +1017,147 @@ type AgentIdentity struct {
...
+	case DoubleType:
+		if warning == nil && criticalValue == -1 {
+			if value.DoubleValue >= criticalValue {
+				return ServiceUnscheduledCritical
+			}
+			return ServiceOk
+		}
+		if critical == nil && (warning != nil && warningValue == -1) {
```
Comment (Hipska) [inline transit/transit.go:1124]: A service without critical will only become WARNING if it's warning threshold is set to -1?

File: transit/transit.go [hunk 3]
```diff
@@ -1017,3 +1017,147 @@ type AgentIdentity struct {
...
+				return ServiceUnscheduledCritical
+			}
+			if (warning != nil && critical != nil) && float64(value.IntegerValue) >= warningValue {
+				return ServiceWarning
+			}
+			return ServiceOk
+		}
+	case DoubleType:
+		if warning == nil && criticalValue == -1 {
+			if value.DoubleValue >= criticalValue {
```
Comment (Hipska) [inline transit/transit.go:1119]: A service without warning will only become CRITICAL if it's critical threshold is set to -1?

File: transit/transit.go [hunk 4]
```diff
@@ -1017,3 +1017,147 @@ type AgentIdentity struct {
...
+			}
+			return ServiceOk
+		}
+		if critical == nil && (warning != nil && warningValue == -1) {
+			if value.DoubleValue >= warningValue {
+				return ServiceWarning
+			}
+			return ServiceOk
+		}
+		if (warning != nil && critical != nil) && (warningValue == -1 || criticalValue == -1) {
```
Comment (Hipska) [inline transit/transit.go:1130]: If one of the thresholds is -1 then service is always OK?

File: transit/transit.go [hunk 5]
```diff
@@ -1017,3 +1017,147 @@ type AgentIdentity struct {
...
+		}
+		// is it a reverse comparison (low to high)
+		if warningValue > criticalValue {
+			if value.DoubleValue <= criticalValue {
+				return ServiceUnscheduledCritical
+			}
+			if value.DoubleValue <= warningValue {
+				return ServiceWarning
+			}
+			return ServiceOk
```
Comment (Hipska) [inline transit/transit.go:1141]: Can be left out

File: transit/transit.go [hunk 6]
```diff
@@ -1017,3 +1017,147 @@ type AgentIdentity struct {
...
+		} else {
+			if value.DoubleValue >= criticalValue {
+				return ServiceUnscheduledCritical
+			}
+			if value.DoubleValue >= warningValue {
+				return ServiceWarning
+			}
+			return ServiceOk
```
Comment (Hipska) [inline transit/transit.go:1149]: can be left out

File: transit/transit.go [hunk 7]
```diff
@@ -1017,3 +1017,147 @@ type AgentIdentity struct {
...
+		if metric.Thresholds != nil {
+			var warning, critical ThresholdValue
+			for _, threshold := range *metric.Thresholds {
+				switch threshold.SampleType {
+				case Warning:
+					warning = threshold
+				case Critical:
+					critical = threshold
+				default:
+					return ServiceOk, fmt.Errorf("unsupported threshold Sample type")
```
Comment (Hipska) [inline transit/transit.go:1043]: As said before, I would expect UNKNOWN in this case.

**Review by Hipska — COMMENTED** [review]
Status calculation has some issues..

**Pull request merged**